notebooks/Tutorial8-Time_series.py

```diff
@@ -0,0 +1,287 @@
+# - `example_ts2.csv`: time series data for `sensor3`.
+
+vanilla_df = pd.read_csv("../fixtures/example_vanilla.csv")
+ts1_df = pd.read_csv("../fixtures/example_ts1.csv")
+ts2_df = pd.read_csv("../fixtures/example_ts2.csv")
+
+vanilla_df.head()
+
+# **Note that the vanilla data contains a column `id` required to link time series data point with each individual.**
+
+ts1_df
+
+ts2_df
+
+# ## Overview of data
+
+# We provide below a basic visualization function to better understand how the time series data look like.
+
+# +
+from typing import Optional, Tuple
+import matplotlib.pyplot as plt
+import matplotlib
+
+
+def plot_series(
+    df: pd.DataFrame,
+    variable_to_plot: str,
+    id_variable: str,
+    time_variable: str,
+    proportion_to_plot: float = 1.0,
+    n_series_to_plot: Optional[int] = None,
+    figsize: Tuple[int, int] = (14, 8),
+) -> matplotlib.figure.Figure:
+    """Plot given series."""
+    if n_series_to_plot is None:
+        n_series_to_plot = df[id_variable].unique().shape[0]
+
+    df_tmp = df.copy()
+    df_tmp = df_tmp.sort_values(by=[id_variable, time_variable]).reset_index(drop=True)
+
+    cmap = matplotlib.cm.get_cmap("gist_rainbow")
+
+    fig, ax = plt.subplots(figsize=figsize)
+    for id_nb, id_name in enumerate(set(df_tmp[id_variable])):
+        if id_nb > n_series_to_plot:
+            break
+        selected_records = df_tmp[df_tmp[id_variable] == id_name]
+        n_points = int(proportion_to_plot * len(selected_records))
+        selected_indices = np.linspace(0, len(selected_records) - 1, num=n_points)
+
+        x = selected_records[time_variable].iloc[selected_indices]
+        y = selected_records[variable_to_plot].iloc[selected_indices]
+        ax.plot(x, y, color=cmap(id_nb / n_series_to_plot))
+    ax.set_title(f"Series for variable {variable_to_plot}")
+    return fig
+
+
+# -
+
+# and we use it to plot sensor data for some individuals
+
+# ### `sensor1` data
+
+plot = plot_series(
+    df := ts1_df,
+    variable_to_plot="sensor1",
+    id_variable="id",
+    time_variable="t",
+    proportion_to_plot=1.0,
+    n_series_to_plot=None,
+    figsize=(14, 8),
+)
+plot.show()
+
+# ###  `sensor2` data
+
+plot = plot_series(
+    df := ts1_df,
+    variable_to_plot="sensor2",
+    id_variable="id",
+    time_variable="t",
+    proportion_to_plot=1.0,
+    n_series_to_plot=None,
+    figsize=(14, 8),
+)
+plot.show()
+
+# ### `sensor3` data
+
+plot = plot_series(
+    df := ts2_df,
+    variable_to_plot="sensor3",
+    id_variable="id",
+    time_variable="t",
+    proportion_to_plot=1.0,
+    n_series_to_plot=None,
+    figsize=(14, 8),
+)
+plot.show()
+
+# Now that we know what data we are maniplating, we can anonymize it.
+
+# ## Avatars of time series data
+
+# ### Upload data and save datasets
+
+# +
+dataset_vanilla = client.pandas_integration.upload_dataframe(vanilla_df)
```

Comment:
don't we need to patch the id varible as `id_column`

Reply:
not required